src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,231 @@
+     * Gets the total number of tasks in the task list.
+     *
+     * @return The total number of tasks in the task list.
+     */
+    public static int getTaskCount() {
+        return tasks.size();
+    }
+
+    /**
+     * Generates a list of string representations of tasks meant for display.
+     * Intended to be used by Ui for displaying all tasks.
+     *
+     * @return A list of strings that represent task information to be displayed.
+     */
+    public static List<String> getTasksStrings() {
+        ArrayList<String> tasksStrings = new ArrayList<>();
+
+        for (int i = 0; i < tasks.size(); i++) {
+            tasksStrings.add(tasks.get(i).toString());
+        }
+
+        return tasksStrings;
+    }
+
+    /**
+     * Adds the given task to the list.
+     *
+     * @param task The task to add to the list.
+     * @return The task that was added.
+     */
+    public static Task addToList(Task task) {
+        tasks.add(task);
+
+        return task;
+    }
+
+    /**
+     * Creates a ToDo task from the given argument string.
+     *
+     * @param args The argument string to be parsed.
+     * @return The task that was added.
+     */
+    public static Task addToDo(String args) throws DukeException {
+        String name = args.strip();
+        if (name == "") {
+            throw new DukeException("Failed to create todo: No task name given");
+        }
+
+        return addToList(new ToDo(name, false));
+    }
+
+    /**
+     * Creates a Deadline task from the given argument string.
+     *
+     * @param args The argument string to be parsed.
+     * @return The task that was added.
+     */
+    public static Task addDeadline(String args) throws DukeException {
+        String[] argsArr = args.split(" /by ", 2);
+        if (argsArr.length < 2) {
+            throw new DukeException("Failed to create deadline: Invalid number of arguments");
+        }
+
+        String name = argsArr[0].strip();
+        String dateStr = argsArr[1].strip();
+
+        if (name == "") {
+            throw new DukeException("Failed to create deadline: No task name given");
+        }
+
+        if (dateStr == "") {
+            throw new DukeException("Failed to create deadline: No date given");
+        }
+
+        LocalDate date;
+        try {
+            date = LocalDate.parse(dateStr);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Failed to create deadline: Invalid date given");
+        }
+
+        return addToList(new Deadline(name, false, date));
+    }
+
+    /**
+     * Creates an Event task from the given argument string.
+     *
+     * @param args The argument string to be parsed.
+     * @return The task that was added.
+     */
+    public static Task addEvent(String args) throws DukeException {
+        String[] argsArr = args.split(" /at ", 2);
+        if (argsArr.length < 2) {
+            throw new DukeException("Failed to create event: Invalid number of arguments");
```

Comment:
I like how you handle many different types of errors. Good job on error handling!

Reply:
Thank you!